File: devops-scm-api/src/main/java/com/tencent/devops/scm/api/ScmCommand.java
package com.tencent.devops.scm.api;

import com.tencent.devops.scm.api.pojo.repository.ScmProviderRepository;

/**
 * scm命令行接口
 */
public interface ScmCommand {

    /**
     * 获取远端仓库信息
     * 主要用户校验凭证有效性
     * @param repository 提供者代码库
     */
    void remoteInfo(ScmProviderRepository repository);
}

File: devops-scm-api/src/main/kotlin/com/tencent/devops/scm/api/ScmCommand.kt
package com.tencent.devops.scm.api

import com.tencent.devops.scm.api.pojo.repository.ScmProviderRepository

/**
 * SCM命令行接口
 */
interface ScmCommand {
    /**
     * 获取远端仓库信息
     * 主要用于校验凭证有效性
     * @param repository 提供者代码库信息
     */
    fun remoteInfo(repository: ScmProviderRepository)
}
